libs/wingsdk/src/target-sim/function.inflight.ts

```diff
@@ -86,7 +89,6 @@ export class Function implements IFunctionClient, ISimulatorResourceInstance {
           timeout: this.timeout,
           context: {
             process: sandboxProcess,
-            $simulator: this.context,
```

Comment:
What kind of opt-in mechanism do you have in mind?

It's probably doable (maybe e.g. we add a parameter to `toInflight()` opt into a "network safe" version of the client), I just wasn't sure if the optimization and increased API surface area was worth it. (For what it's worth, we already spin up an HTTP server per website, API, and public bucket in your Wing app 😅). I'll double check perf numbers tomorrow.

re: big picture -- some other benefits are that this change eliminates some of the differences between the simulator clients and actual-cloud clients (they're all doing HTTP requests now, and are susceptible to latency and network errors), and it paves the way for making simulated resources available in other languages.

Reply:
I was thinking more of an implicit opt-in. The simulator client can check for the existence of a $simulator global, and only go with HTTP if that global isn't found.

It would be cool to keep the same API and always use JSON as the way to transport data, regardless of using HTTP or calling directly. This keeps things that matter the same imo.

> eliminates some of the differences between the simulator clients and actual-cloud clients

Arguably I think we're willing to have differences for non-functional things (e.g. latency and networking hiccups). I do think requiring serialization is a reasonable functional change keep the same between them.

Reply:
For now I'd like to at least try out using the HTTP everywhere to see if it works / is reliable. The main upside in my mind is to avoid bugs that only reproduce when the client is in a different process. All that said I'm totally fine changing things if it ends up creating issues for contributing or has performance drawbacks (or if this implementation falls flat on its face 🙃 - which is totally possible)